content/combinatorial/IntPerm.h

```diff
@@ -10,7 +10,7 @@
 
 int permToInt(vi& v) {
 	int use = 0, i = 0, r = 0;
-	trav(x, v) r = r * ++i + __builtin_popcount(use & -(1 << x)),
+	for(auto x:v) r = r * ++i + __builtin_popcount(use&-(1<<x)),
```

Comment:
int to save a character which can be spent on whitespace.

auto is good when the type could be replaced by something else by the user, but in this case only int fits. (I made the same comment in a few other places, but didn't go through all files)